### PR TITLE
Add support for OTEL HTTP exporter

### PR DESCRIPTION
**Pull request by rofafor**
Works with Elastic: `tracerprovider.WithHTTPExporter("apmserver:443", "Bearer foobar",  false)`

### CODE REVIEW

File: v2/service/module/tracerprovider/tracerprovider.go [hunk 1]
```diff
@@ -128,29 +167,66 @@ func WithSamplePercentage(percentage int) Opt {
 	}
 }
 
+// WithCollector is deprecated. Use WithGRPCExporter instead.
 func WithCollector(host string, port int, credentials credentials.TransportCredentials) Opt {
```
Comment (rofafor) [inline v2/service/module/tracerprovider/tracerprovider.go:171]: Kept this due to backwards compatibility - can be removed as well.

File: v2/service/module/tracerprovider/tracerprovider.go [hunk 2]
```diff
@@ -128,29 +179,66 @@ func WithSamplePercentage(percentage int) Opt {
...
+// WithHTTPExporter sets HTTP exporter endpoint for trace exporter with an authorization token.
+// Endpoint is "host:port" of to the collector and you can set token as "ApiKey <token>" or "Bearer <token>".
+// Set token to empty if not needed. Insecure disables TLS.
+func WithHTTPExporter(endpoint string, token string, insecure bool) Opt {
 	return func(tp *TracerProvider) error {
-		tp.collectorHost = host
-		tp.collectorPort = port
-		tp.credentials = credentials
+		if token != "" && !(strings.HasPrefix(strings.ToLower(token), "apikey ") || strings.HasPrefix(strings.ToLower(token), "bearer ")) {
+			return ErrInvalidToken
```
Comment (zetaab) [inline v2/service/module/tracerprovider/tracerprovider.go:205]: just thinking is it valid to return error if someone have prefix outside of apikey/bearer? Or should we print warning to logs instead?

Comment (rofafor) [reply v2/service/module/tracerprovider/tracerprovider.go:205]: Are there actually something else? E.g. Elastic APM supports only these two: https://www.elastic.co/guide/en/observability/current/apm-open-telemetry-direct.html#apm-connect-open-telemetry-collector

Comment (zetaab) [reply v2/service/module/tracerprovider/tracerprovider.go:205]: well, lets start with that

**Pull request merged**